go/merge/three_way_test.go

```diff
@@ -0,0 +1,178 @@
+// Copyright 2016 Attic Labs, Inc. All rights reserved.
+// Licensed under the Apache License, version 2.0:
+// http://www.apache.org/licenses/LICENSE-2.0
+
+package merge
+
+import (
+	"testing"
+
+	"github.com/attic-labs/noms/go/types"
+	"github.com/attic-labs/testify/assert"
+)
+
+var (
+	aa1      = createMap("a1", "a-one", "a2", "a-two", "a3", "a-three", "a4", "a-four")
+	aa1a     = createMap("a1", "a-one", "a2", "a-two", "a3", "a-three-diff", "a4", "a-four", "a6", "a-six")
+	aa1b     = createMap("a1", "a-one", "a3", "a-three-diff", "a4", "a-four", "a5", "a-five")
+	aaMerged = createMap("a1", "a-one", "a3", "a-three-diff", "a4", "a-four", "a5", "a-five", "a6", "a-six")
+
+	mm1       = createMap()
+	mm1a      = createMap("k1", createMap(0, "a"))
+	mm1b      = createMap("k1", createMap(1, "b"))
+	mm1Merged = createMap("k1", createMap(0, "a", 1, "b"))
+
+	mm2       = createMap("k2", aa1, "k3", "k-three")
+	mm2a      = createMap("k1", createMap(0, "a"), "k2", aa1a, "k3", "k-three", "k4", "k-four")
+	mm2b      = createMap("k1", createMap(1, "b"), "k2", aa1b)
+	mm2Merged = createMap("k1", createMap(0, "a", 1, "b"), "k2", aaMerged, "k4", "k-four")
+)
+
+func TestThreeWayMergeMap_DoNothing(t *testing.T) {
+	assert := assert.New(t)
+
+	merged, err := ThreeWay(nil, nil, aa1, nil)
+	if assert.NoError(err) {
+		assert.Equal(aa1, merged)
+	}
+}
+
+func TestThreeWayMergeMap_NoRecursion(t *testing.T) {
+	assert := assert.New(t)
+
+	merged, err := ThreeWay(aa1a, aa1b, aa1, nil)
+	if assert.NoError(err) {
+		assert.Equal(aaMerged, merged)
+	}
+}
+
+func TestThreeWayMergeMap_RecursiveCreate(t *testing.T) {
+	assert := assert.New(t)
+
+	merged, err := ThreeWay(mm1a, mm1b, mm1, nil)
```

Comment:
I think it would be valuable for these tests to assert both `ThreeWay(a, b, parent)` and `ThreeWay(b, a, parent)` ... they should give the same results, right?

Seems like you could fairly easily have a function like `testThreeWay` and similar for errors which does both.